janitor/functions/concatenate_columns.py

```diff
@@ -51,8 +51,12 @@ def concatenate_columns(
     if len(column_names) < 2:
         raise JanitorError("At least two columns must be specified")
 
+    df = df.copy()  # avoid mutating original data
```

Comment:
Now that we've added this, the function is no longer mutating.Can we also update the docstring?